### PR TITLE
Add $ http-request to testdrive

### PR DESCRIPTION
**Pull request by philip-stoev**
This is needed in order to support curl-like HTTP requests against
Debezium for administration purposes

### CODE REVIEW

File: src/testdrive/src/action/http.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    async fn undo(&self, _: &mut State) -> Result<(), String> {
+        Ok(())
+    }
+
+    async fn redo(&self, _: &mut State) -> Result<(), String> {
+        println!("$ http-request {} {}\n{}", self.method, self.url, self.body);
+
+        let client = reqwest::Client::new();
+
+        let response = match self.method.as_ref() {
```
Comment (petrosagg) [inline src/testdrive/src/action/http.rs:44]: What do you think of using `reqwest::Method` to parse the method and open up the possibility for any valid method? Something like:

```rust
        let method = self
            .method
            .parse()
            .or_else(|_| Err(format!("Unknown http method type: {}", self.method)))?;

        let mut request = client.request(method, &self.url).body(self.body.clone());

        if let Some(value) = &self.content_type {
            request = request.header(CONTENT_TYPE, value);
        }

        let response = request.send().await.map_err(|e| e.to_string())?;
```

File: src/testdrive/src/action/http.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            "GET" => client.get(&self.url),
+            "POST" => client
+                .post(&self.url)
+                .header(
+                    CONTENT_TYPE,
+                    self.content_type
+                        .as_ref()
+                        .ok_or_else(|| "content-type requred for POST requests")?,
```
[the remark this answers is not in the document]
Comment (petrosagg) [reply src/testdrive/src/action/http.rs:52]: also, it would be nice to allow this header for any method

**Review by benesch — APPROVED** [review]
LGTM modulo Petros's comments! Tyvm!

File: src/testdrive/src/action/http.rs (hunk 1 above)
Comment (philip-stoev) [reply src/testdrive/src/action/http.rs:44]: done.

File: src/testdrive/src/action/http.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (philip-stoev) [reply src/testdrive/src/action/http.rs:52]: done

**Pull request merged**